build.ps1

```diff
@@ -96,36 +96,157 @@ if (!(Get-Command "docker" -ErrorAction SilentlyContinue))
 }
 
 if (!$Tag)
 {
-    if (Test-Path "$ImageDirectory/metadata")
+    if ($env:GITHUB_ACTIONS -eq "true")
     {
-        $Tag = "-t $DockerOrg/$Name"
-        $Version = Get-Content "$ImageDirectory/metadata/IMAGE_VERSION"
-        $Tag += ":$Version"
-        $Revision = Get-Content "$ImageDirectory/metadata/IMAGE_REVISION"
+        $Tag = "$DockerOrg/${Name}:$Version"
+        $Revision = Get-Content (Join-Path $ImageDirectory "metadata" "IMAGE_REVISION")
         if ($Revision)
         {
             $Tag += "-$Revision"
         }
-        $Tag += " $(Get-Content $ImageDirectory/metadata/ADDITIONAL_TAGS | ForEach-Object { $_.replace("$Name","$DockerOrg/$Name") })"
+        $AdditionalTags = "$(Get-Content (Join-Path $ImageDirectory "metadata" "ADDITIONAL_TAGS") | ForEach-Object { $_.replace("$Name","$DockerOrg/$Name") })"
+        Write-Host "If pushed, the image will be available as: $Tag $AdditionalTags"
     }
     else
     {
-        throw "No metadata found for $Name"
+        $Tag = "$DockerOrg/${Name}:dev"
+        $AdditionalTags = ""
+        Write-Host "The image will be locally runnable as:" $Tag
     }
 }
 else
 {
     Write-Host "Tag value set by script parameter:" $Tag
+    $AdditionalTags = ""
 }
 
-$docker_command = "docker build $Tag $ImageDirectory"
-Write-Host $docker_command
-Invoke-Expression $docker_command
+if ($Name -eq "uaa-server")
+{
+    $Dockerfile = Join-Path $ImageDirectory Dockerfile
+    if (!(Test-Path $Dockerfile))
+    {
+        throw "No Dockerfile for $Name (expected $Dockerfile)"
+    }
+
+    $docker_command = "docker build -t $Tag $AdditionalTags $ImageDirectory --build-arg SERVER_VERSION=$Version"
+    Write-Host $docker_command
+    Invoke-Expression $docker_command
+}
+else
+{
+    if (!(Get-Command "patch" -ErrorAction SilentlyContinue))
+    {
+        if (Test-Path "$Env:ProgramFiles\Git\usr\bin\patch.exe")
+        {
+            Write-Host "'patch' command not found, but Git is installed; adding Git usr\bin to PATH"
+            $env:Path += ";$Env:ProgramFiles\Git\usr\bin"
+        }
+        else
+        {
+            throw "'patch' command not found"
+        }
+    }
+
+    switch ($Name)
+    {
+        "config-server"
+        {
+            $appName = "ConfigServer"
+            $dependencies = "cloud-config-server,actuator,cloud-eureka,security"
+        }
+        "eureka-server"
+        {
+            $appName = "EurekaServer"
+            $dependencies = "cloud-eureka-server,actuator"
+        }
+        "spring-boot-admin"
+        {
+            $appName = "SpringBootAdmin"
+            $dependencies = "codecentric-spring-boot-admin-server,native"
+        }
+        Default
+        {
+            Write-Host "$Name is not currently supported by this script"
+            exit 2
+        }
+    }
+
+    $workPath = "workspace"
+    if (!(Test-Path $workPath))
+    {
+        New-Item -ItemType Directory -Path $workPath | Out-Null
+    }
+    Set-Location $workPath
```

Comment:
It would be cleaner to use `Push-Location` and `Pop-Location`.

https://stackoverflow.com/questions/32734408/when-to-use-push-location-versus-set-location

Ideally, in a try/catch, because when I terminate the script, it breaks inside the workspace/name subdirectory.